denlillemand/week2/index.html

```diff
@@ -0,0 +1,299 @@
+<html>
+<head>
+	<title>Hello world with react</title>
+	<style type="text/css">
+	body {
+		width: 1000px;
+		margin: auto;
+	}
+	</style>
+
+	<script src="http://fb.me/react-0.14.7.min.js"></script>
+	<script src="http://fb.me/react-dom-0.14.7.min.js"></script>
+	<script src="https://cdnjs.cloudflare.com/ajax/libs/babel-core/5.8.23/browser.min.js"></script>
+</head>
+<body>
+    	<div id="root">
+
+	</div>    
+
+	<script type="text/babel">
+	    
+	    class NamedListForm extends React.Component {
+		constructor(props, context) {
+		    super(props,context);
+		    this.state = {
+			name: "",
+		    };
+		    this.onChange = this.onChange.bind(this);
+		    this.onSubmit = this.onSubmit.bind(this);
+		}
+
+		onChange(event) {
+		    this.setState({
+			name: event.target.value
+		    });
+		}
+
+		onSubmit(event) {
+		    this.props.onSubmitNamedList(this.state.name);
+		    this.setState({
+			name: ""
+		    });
+		}
+
+		render() {
+		    return(
+		    	<div>
+			    <input value={this.state.name} onChange={this.onChange} />
+			    <button onClick={this.onSubmit}>Submit list</button>
+			</div>
+		    );
+		}
+	    } 
+
+
+	    class Message extends React.Component {
+		constructor(props, context) {
+		    super(props, context);
+		    this.state = {};
+		    this.deleteMessage = this.deleteMessage.bind(this);
+		    this.archive = this.archive.bind(this);
+		}
+
+		deleteMessage(event) {
+		    this.props.deleteMessage(this.props.message.id);
+		}
+
+		archive(event) {
+		    let message = this.props.message;
+		    this.props.archive(message.id, message.archived);
+		}
+
+		render() {
+		    return(
+		    	<div>
+			    	{this.props.message.id}:
+				{this.props.message.text}:
+				{this.props.message.archived ? "ARCHIVED" : "NOT ARCHIVED"}
+				<button onClick={this.archive}>{this.props.message.archived ? "Unarchive" : "Archive"}</button>
+				<button onClick={this.deleteMessage}>Delete</button>
+			</div>
+		    
+		    );
+		}
+	    }
+	    Message.propTypes = {
+		deleteMessage: React.PropTypes.func,
+		message: React.PropTypes.object,
+		archive: React.PropTypes.func
+	    }
+
+	    class MainList extends React.Component {
+		constructor(props, context) {
+		    super(props, context);
+		    this.state = {};
+		    this.swapMessage = this.swapMessage.bind(this);
+		    this.archiveMessage = this.archiveMessage.bind(this);
+		    this.onSubmitNamedList = this.onSubmitNamedList.bind(this);
+		    this.deleteMessage = this.deleteMessage.bind(this);
+		    this.onSubmitMessage = this.onSubmitMessage.bind(this);
+		}
+
+		archiveMessage(id, currentList, isArchived) {
+		    this.state[currentList].forEach(function(message) {
+			if(message.id === id) {
+			    message.archived = !isArchived;
```

Comment:
you are directly mutating state here. You should first clone the object (see  `Object.assign`), mutate the clone and then update the state.

```javascript
var mutatedArray = this.state[currentList].map(function(message) {
    if (message.id === id) {
      var clone = Object.assign({}, message);
      clone .archived = !isArchived;
      return clone;
    }
    return message;
  })
this.setState({
  [currentList]: mutatedArray
})
```